components/TicTacToe.jsx

```diff
@@ -7,22 +7,64 @@ export const TicTacToe = () => {
     border: "1px solid",
     boxSizing: "content-box",
   };
+  const cols = 3;
+  const rows = 3;
   const pieces = ["X", "O"];
   const [turn, setTurn] = useState(0);
   const [grid, setGrid] = useState(Array(9).fill(null));
+  const [win, setWin] = useState("in progress");
+
+  const notClicked = (i) => {
+    if (pieces.indexOf(grid[i]) < 0) {
+      return true;
+    }
+    return false;
+  };
+
+  const gridIndex = (x, y) => {
+    return cols * x + y;
+  };
+
+  const checkWinHorizontal = () => {
+    for (let x = 0; x < rows; x += 1) {
+      for (let y = 0; y < cols; y += 1) {
```

Comment:
```suggestion
    for (let x = 0; x < rows; ++x) {
      for (let y = 0; y < cols; ++y) {
```

Prefix increment is more elegant and performant

See: https://stackoverflow.com/a/484572/865091